src/saving/SaveList.cs

```diff
@@ -207,12 +216,40 @@ public void Refresh()
 
         savesList.QueueFreeChildren();
 
+        allItemsCount = 0;
+        loadedItems = 0;
+
         loadingItem.Visible = true;
         readSavesList = new Task<List<string>>(() => SaveHelper.CreateListOfSaves());
         TaskExecutor.Instance.AddTask(readSavesList);
         EmitSignal(nameof(OnItemsChanged));
     }
 
+    public void Filter(SaveInformation.SaveType? saveTypeFilter)
+    {
+        foreach (var item in savesList.GetChildren())
```

Comment:
GetChildren should be avoided, the style in Thrive is to store child nodes in a list. A new field like `List<SaveListItem> savesListItems` needs to be added, I didn't originally add such a thing as it wasn't necessary as the list didn't need to care about its child items in the past.